phpstan.neon

```diff
@@ -606,3 +606,9 @@ parameters:
         -
             message: '#Function "(class_exists|interface_exists)\(\)" cannot be used/left in the code\: use ReflectionProvider\->has\*\(\) instead#'
             path: packages/Skipper/SkipCriteriaResolver/SkippedClassResolver.php
+
+        # todo: to be updated to use NewOptimizedDirectorySourceLocator later
+        # currently changing to NewOptimizedDirectorySourceLocator cause error, @see https://github.com/rectorphp/rector-src/actions/runs/5965685296/job/16183665877#step:10:19
+        -
+            message: '#Instantiation of deprecated class PHPStan\\Reflection\\BetterReflection\\SourceLocator\\OptimizedDirectorySourceLocator#'
+            path: packages/NodeTypeResolver/Reflection/BetterReflection/SourceLocatorProvider/DynamicSourceLocatorProvider.php
```

Comment:
this new `NewOptimizedDirectorySourceLocator` is a todo later as currently cause error when applied, noted above.

Reply:
@TomasVotruba ok, it seems the error stil exists, so the issue is not because of `NewOptimizedDirectorySourceLocator`

https://github.com/rectorphp/rector-src/actions/runs/5965825544/job/16184110293#step:10:19

Reply:
@TomasVotruba let's revert the revert, I will look into the fix :)